src/formElementPlugins/number/src/ApplicationView.tsx

```diff
@@ -0,0 +1,169 @@
+    step = 1,
+    locale = undefined,
+    currency = undefined,
+    prefix = '',
+    suffix = '',
+    maxSignificantDigits = undefined,
+  } = parameters
+  const [textValue, setTextValue] = useState<string | null | undefined>(currentResponse?.text)
+  const [internalValidation, setInternalValidation] = useState(validationState)
+
+  const formatOptions = {
+    style: currency ? 'currency' : undefined,
+    currency: currency !== '' ? currency : undefined,
+    maximumSignificantDigits: maxSignificantDigits,
+  }
+  const numberFormatter = new Intl.NumberFormat(locale, formatOptions)
+
+  useEffect(() => {
+    // Ensures defaultValue is saved on first load
+    if (!currentResponse?.text && defaultValue !== undefined) {
+      const [number, text] = parseInput(
+        String(defaultValue),
+        numberFormatter,
+        simple,
+        prefix,
+        suffix
+      )
+      const validation = customValidate(number, type, minValue, maxValue)
+      setInternalValidation(validation)
+      if (validation.isValid) onSave({ text, number, type, currency, locale })
+      setTextValue(text)
+    }
+  }, [])
+
+  useEffect(() => {
+    if (!textValue) return
+    handleLoseFocus()
+  }, [locale, minValue, maxValue, currency, prefix, suffix, maxSignificantDigits, type])
+
+  function handleChange(e: any) {
+    const text = e.target.value
+    const [number, _] = parseInput(text, numberFormatter, simple, prefix, suffix)
+    setInternalValidation(customValidate(number, type, minValue, maxValue))
+    onUpdate(text)
+    setTextValue(text)
+  }
+
+  function handleLoseFocus() {
+    const [number, text] = parseInput(textValue, numberFormatter, simple, prefix, suffix)
+    const validation = customValidate(number, type, minValue, maxValue)
+    setInternalValidation(validation)
+    if (validation.isValid) onSave({ text, number, type, currency, locale })
+    else onSave(null)
+    setTextValue(text)
+  }
+
+  const styles = maxWidth
+    ? {
+        maxWidth,
+      }
+    : {}
+
+  return (
+    <>
+      <label>
+        <Markdown text={label} semanticComponent="noParagraph" />
+      </label>
+      <Markdown text={description} />
+      <Form.Input
+        fluid
+        type={simple ? 'number' : 'text'}
+        min={minValue}
+        max={maxValue}
+        step={step}
+        placeholder={placeholder}
+        onChange={handleChange}
+        onBlur={handleLoseFocus}
+        onFocus={setIsActive}
+        value={textValue ? textValue : ''}
+        disabled={!isEditable}
+        style={styles}
+        error={
+          !validationState.isValid || !internalValidation.isValid
+            ? {
+                content:
+                  internalValidation?.validationMessage || validationState?.validationMessage,
+                pointing: 'above',
+              }
+            : null
+        }
+      />
+    </>
+  )
+}
+
+const parseInput = (
+  textInput: string | null | undefined,
+  numberFormatter: any,
```

Comment:
Maybe good to use a type here? I was wondering would this fail if no `numberFormat` is passed?

Reply:
I agree -- any idea how to figure out what the "type" should be?

> I was wondering would this fail if no `numberFormat` is passed?

That won't happen, since numberFormatter is constructed at the top of the module, and uses default values if appropriate parameters are not provided.

Reply:
> I agree -- any idea how to figure out what the "type" should be?

Got it! bff32532